src/consent-string.js

```diff
@@ -0,0 +1,343 @@
+const { encodeConsentString } = require('./encode');
+const { decodeConsentString } = require('./decode');
+
+/**
+ * Regular expression for validating
+ */
+const consentLanguageRegexp = /^[a-z]{2}$/;
+
+class ConsentString {
+  /**
+   * Initialize a new ConsentString object
+   *
+   * @param {string} baseString An existing consent string to parse and use for our initial values
+   */
+  constructor(baseString = null) {
+    this.created = new Date();
+    this.lastUpdated = new Date();
+
+    /**
+     * Version number of consent string specification
+     *
+     * @type {integer}
+     */
+    this.version = 1;
+
+    /**
+     * Vendor list with format from https://github.com/InteractiveAdvertisingBureau/GDPR-Transparency-and-Consent-Framework/blob/master/Draft_for_Public_Comment_Transparency%20%26%20Consent%20Framework%20-%20cookie%20and%20vendor%20list%20format%20specification%20v1.0a.pdf
+     *
+     * @type {object}
+     */
+    this.vendorList = null;
+
+    /**
+     * Version of the vendor list used for the purposes and vendors
+     *
+     * @type {integer}
+     */
+    this.vendorListVersion = null;
+
+    /**
+     * The unique ID of the CMP that last modified the consent string
+     *
+     * @type {integer}
+     */
+    this.cmpId = null;
+
+    /**
+     * Version of the code used by the CMP when collecting consent
+     *
+     * @type {integer}
+     */
+    this.cmpVersion = null;
+
+    /**
+     * ID of the screen used by CMP when collecting consent
+     *
+     * @type {integer}
+     */
+    this.consentScreen = null;
+
+    /**
+     * Two-letter ISO639-1 code (en, fr, de, etc.) of the language that the CMP asked consent in
+     *
+     * @type {string}
+     */
+    this.consentLanguage = null;
+
+    /**
+     * List of purpose IDs that the user has given consent to
+     *
+     * @type {integer[]}
+     */
+    this.allowedPurposeIds = [];
+
+    /**
+     * List of vendor IDs that the user has given consent to
+     *
+     * @type {integer[]}
+     */
+    this.allowedVendorIds = [];
+
+    // Decode the base string
+    if (baseString) {
+      Object.assign(this, decodeConsentString(baseString));
+    }
+  }
+
+  /**
+   * Get the web-safe, base64-encoded consent string
+   *
+   * @return {string} Web-safe, base64-encoded consent string
+   */
+  getConsentString(updateDate = true) {
+    if (!this.vendorList) {
+      throw new Error('ConsentString - A vendor list is required to encode a consent string');
+    }
+
+    if (updateDate === true) {
```

Comment:
Maybe just `if (updateDate) {} ` instead of explicitly checking for a value equal to `true`